icu4c/source/i18n/messageformat2_errors.h

```diff
@@ -66,6 +66,15 @@ namespace message2 {
     enum DynamicErrorType {
         UnresolvedVariable,
         FormattingError,
+        BadOptionError,
```

Comment:
If these are meant as documentation, it would be better to put them where Doxygen can find them, not just when scrolling through.

```suggestion
        BadOptionError, //< Used to signal…
```

or

```suggestion
        /** Used to signal … */
        BadOptionError,
```

Reply:
Fixed in [e525cac](https://github.com/unicode-org/icu/pull/3423/commits/e525cacb99e2185ec4e6a10965015c84004f80d8)